pkg/api/dirs.go

```diff
@@ -144,10 +148,57 @@ func storeDir(ctx context.Context, reader io.ReadCloser, s storage.Storer, logge
 		contentType: ManifestContentType,
 		reader:      r,
 	}
-	manifestReference, err := storeFile(ctx, manifestFileInfo, s)
+	manifestReference, err := storeFile(ctx, manifestFileInfo, s, mode)
 	if err != nil {
 		return swarm.ZeroAddress, fmt.Errorf("store manifest error: %w", err)
 	}
 
 	return manifestReference, nil
 }
+
+// storeFile uploads the given file and returns its reference
+// this function was extracted from `fileUploadHandler` and should eventually replace its current code
+func storeFile(ctx context.Context, fileInfo *fileUploadInfo, s storage.Storer, mode storage.ModePut) (swarm.Address, error) {
+	v := ctx.Value(toEncryptContextKey{})
+	toEncrypt, _ := v.(bool) // default is false
+
+	// first store the file and get its reference
+	sp := splitter.NewSimpleSplitter(s, mode)
+	fr, err := file.SplitWriteAll(ctx, sp, fileInfo.reader, fileInfo.size, toEncrypt)
+	if err != nil {
+		return swarm.ZeroAddress, fmt.Errorf("split file error: %w", err)
```

Comment:
Word error is redundant it the error message. This goes for other errors in this function.